Add support for zwave-js 12.0 and schema 32

raman325 opened the pull request:
## Breaking Changes

Per the [Z-Wave JS 12.x migration guide](https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v12), many commands in the driver have been renamed, and the corresponding model methods/properties and events have changed accordingly:
- `Controller.is_heal_network_active` property → `Controller.is_rebuilding_routes`
- `Controller.async_begin_healing_network` method → `Controller.async_begin_rebuilding_routes`
- `Controller.async_stop_healing_network` method → `Controller.async_stop_rebuilding_routes`
- `Controller.async_heal_node` method → `Controller.async_rebuild_node_routes`
- `heal network progress` controller event -> `rebuild routes progress`
- `heal network done` controller event -> `rebuild routes done`

Additionally, `Controller.async_firmware_update_ota` now requires `list[NodeFirmwareUpdateInfo]` instead of `list[NodeFirmwareUpdateFileInfo]` as an update

## Description

Follows https://github.com/zwave-js/zwave-js-server/pull/1036

Migration guide: https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v12

MartinHjelmare reviewed (commented):
Can we add tests for the non covered lines?

raman325 commented:
> Can we add tests for the non covered lines?

Once the PR linked above gets merged, I will refactor this one and add the missing coverage, we should be able to get this PR to 100%

MartinHjelmare commented:
Please mention the breaking changes in the client in the PR description.

MartinHjelmare commented on zwave_js_server/exceptions.py:
```diff
@@ -86,7 +86,7 @@ def __init__(
     ) -> None:
         """Initialize a failed command error."""
-        super().__init__(msg or f"Command failed: {error_code}")
+        super().__init__(f"{error_code}: {msg}" or f"Command failed: {error_code}")
```
We can't write the condition like that now since the first part is always true.

raman325 replied:
good point

raman325 commented:
> Please mention the breaking changes in the client in the PR description.

You mean the fact that the `FailedCommand` exception parameters have changed?

MartinHjelmare commented:
No those haven't changed. We've removed properties and methods and added other ones to replace them regarding healing and changed a parameter in `async_firmware_update_ota` in the controller model.

raman325 commented:
> No those haven't changed. We've removed properties and methods and added other ones to replace them regarding healing and changed a parameter in `async_firmware_update_ota` in the controller model.

ah thanks, thought we were talking about the client module 😅
